llama_index/schema.py

```diff
@@ -119,3 +147,303 @@ def extra_info_str(self) -> Optional[str]:
             return None
 
         return "\n".join([f"{k}: {str(v)}" for k, v in self.extra_info.items()])
+
+
+@dataclass
+class BaseIndexObject(DataClassJsonMixin):
+    """Base Index Object.
+
+    Generic abstract interface for retrievable objects
+
+    """
+
+    _id: str = ""
+    content: Any = None
+    relationships: Dict[DataRelationship, Any] = field(default_factory=dict)
+    embedding: Optional[List[float]] = None
+    object_hash: Optional[str] = None
+    weight: float = 1.0
+
+    """"
+    metadata fields
+    - injected as part of the text shown to LLMs as context
```

Comment:
can we decouple injecting metadata into the text from adding metadata itself? 

adding metadata can still be useful because it can be passed-through to the source nodes for users to lookup/reference. users may want to have control over which metadata keys to inject 

this has always been a pain point, now that we're refactoring this we may as well think about it

Reply:
Yea, maybe we need object content templates?

Reply:
Added some templating stuff, so that it's completely up to the user how it will look like